### ISSUE TITLE
Improve the Playground for the P4 Language

### ISSUE THREAD

**Issue by fruffy** [p4lang/project-ideas#21]
[Scala](https://www.scala-lang.org/) and the [Rust Foundation](https://www.rust-lang.org/) have a Playground where you can try out the language using an interpreter. It would be nice if we had something similar for the P4 language. The switch model can be simple.

**Comment by jnfoster** [comment]
We have this!

https://p4.org/sandbox-page/

**Comment by fruffy** [comment]
Maybe we can make this open-source and maintainable and promote it on the front page?

**Comment by fruffy** [comment]
Oh I see, it is actually on the front-page but I didn't register that.

**Comment by jnfoster** [comment]
We should still get this going with an open-source and up-to-date version. Perhaps using Jaehyun's more actively-maintained reference interpreter.

**Comment by AdarshRawat1** [comment]
We can add a few more standard code examples to the sandbox preview. Currently, it has only one preset.

**Comment by fruffy** [comment]
@jnfoster Do we have access to the source code in some form?

**Comment by jnfoster** [comment]
Yes, of course!

**Comment by fruffy** [comment]
> Yes, of course!

I am guessing this is petr4, but how is it integrated on the website? E.g., adding programs or making changes.

**Comment by jaehyun1ee** [comment]
We have a [web interface](https://kaist-plrg.github.io/p4cherry/) for our OCaml implementation of P4 (based on Petr4). We currently support parts of V1Model and eBPF architectures, and STF tests.

**Comment by fruffy** [comment]
> We have a [web interface](https://kaist-plrg.github.io/p4cherry/) for our OCaml implementation of P4 (based on Petr4). We currently support parts of V1Model and eBPF architectures, and STF tests.

This is nice. I like how it has the architecture supplied too. Is this open-source?

**Comment by jaehyun1ee** [comment]
Yes, this is open-source. It currently operates on our `web-serverless` [branch](https://github.com/kaist-plrg/p4cherry/tree/web-serverless).
